src/NuGet.Core/NuGet.Build.Tasks/GetRestoreSettingsTask.cs

```diff
@@ -0,0 +1,132 @@
+
+        public ISettings GetSettings(string projectDirectory)
+        {
+            if (string.IsNullOrEmpty(RestoreConfigFile))
+            {
+                return Settings.LoadDefaultSettings(projectDirectory,
+                    configFileName: null,
+                    machineWideSettings: _machineWideSettings.Value);
+            }
+            else
+            {
+                var configFileFullPath = Path.GetFullPath(RestoreConfigFile);
+                var directory = Path.GetDirectoryName(configFileFullPath);
+                var configFileName = Path.GetFileName(configFileFullPath);
+                return Settings.LoadSpecificSettings(projectDirectory,
+                    configFileName: configFileName);
+            }
+        }
+
+        public override bool Execute()
+        {
+            // log inputs
+            var log = new MSBuildLogger(Log);
+            log.LogDebug($"(in) ProjectUniqueName '{ProjectUniqueName}'");
+            if (RestoreSources != null)
+            {
+                log.LogDebug($"(in) RestoreSources '{string.Join(";", RestoreSources.Select(p => p))}'");
+            }
+            if (RestorePackagesPath != null)
+            {
+                log.LogDebug($"(in) RestorePackagesPath '{RestorePackagesPath}'");
+            }
+            if (RestoreFallbackFolders != null)
+            {
+                log.LogDebug($"(in) RestoreFallbackFolders '{string.Join(";", RestoreFallbackFolders.Select(p => p))}'");
+            }
+            if (RestoreConfigFile != null)
+            {
+                log.LogDebug($"(in) RestoreConfigFile '{RestoreConfigFile}'");
+            }
+            var settings = GetSettings(Path.GetDirectoryName(ProjectUniqueName));
+
+            if (string.IsNullOrEmpty(RestorePackagesPath))
+            {
+                OutputPackagesPath = SettingsUtility.GetGlobalPackagesFolder(settings);
+            }
+            else
+            {
+                OutputPackagesPath = RestorePackagesPath;
+            }
+
+            if (RestoreSources == null)
+            {
+                var packageSourceProvider = new PackageSourceProvider(settings);
+                var packageSourcesFromProvider = packageSourceProvider.LoadPackageSources();
+                OutputSources = packageSourcesFromProvider.Select(e => e.Source).ToArray();
+            }
+            else
+            {
+                OutputSources = RestoreSources;
+            }
+
+            if (RestoreFallbackFolders == null)
+            {
+                OutputFallbackFolders = SettingsUtility.GetFallbackPackageFolders(settings).ToArray();
+            }
+            else
+            {
+                OutputFallbackFolders = RestoreFallbackFolders;
+            }
+
+            var configFilePaths = new List<string>();
+            foreach (var config in settings.Priority)
+            {
+                configFilePaths.Add(config.Root + config.FileName);
```

Comment:
Path.GetFullPath(Path.Combine(config.Root, config.FileName)) might help here, it would ensure relative paths don't show up.